### DISCUSSION TITLE
同一个定位条件，相同界面ID的两个不同界面都能定位到，这种一般用什么办法区分？

### DISCUSSION THREAD

**Discussion by dearmb** [gkd-kit/gkd#802] [Q&A]
[快照1.zip](https://github.com/user-attachments/files/17985690/1.zip)
[快照2.zip](https://github.com/user-attachments/files/17985691/2.zip)
我用的定位条件是
[vid="touch_intercept_fl"] >n Image[index=1]

**Comment by lisonge** [comment]
它们之前信息差别很大，可以通过很多区别定位，比如添加父节点约束 `View[childCount=2]`

```text
[vid="touch_intercept_fl"] >n View[childCount=2] > Image[index=1]
```

另外可以优化为

```text
View[childCount=2] > @Image[index=1] <<n [vid="touch_intercept_fl"]
```

<https://i.gkd.li/i/17986356?gkd=Vmlld1tjaGlsZENvdW50PTJdID4gQEltYWdlW2luZGV4PTFdIDw8biBbdmlkPSJ0b3VjaF9pbnRlcmNlcHRfZmwiXQ>

**Reply by dearmb** [reply]
[快照3.zip](https://github.com/user-attachments/files/18015683/3.zip)

有啥别的规则吗？比如“定位1并且界面包含xx”

你给的这两条，在另一个界面又冲突了……

**Reply by lisonge** [reply]
属性面板底部有一个图标，点击可以获取当前节点的选择器，你可以直接用那个

**Reply by dearmb** [reply]
![1111](https://github.com/user-attachments/assets/6edb8d7d-0dda-4ac6-a0ba-2410efa85dec)
是指这个吗？

**Comment by dearmb** [comment]
主要就是规则弄复杂一点，不容易重复是吧？
